apache/iotdb
# [IOTDB-4758]Delete snapshot after region migration

## Description

### Content1 ...
Delete remote snapshot after region migration
### test
Test plan:
1. Enable the cluster 1C3D and two replicas
2. kiil drops the process where DataNode2 resides
3. Run the show regions command to view the replication information
4. Use the remove_node.sh script to remove DataNode2
5. Check whether snapshots are generated for the snapshot directories in DataNode1 and DataNode3
6. Wait for a period of time, and check whether the temporary file that completes the replica migration is deleted in DataNode1 and DataNode3. Run the show regions command to check whether the replica migration is correct
7. Connect to the datanode where the migrated replica resides to query data and verify that data is normal

Result:
![image](https://user-images.githubusercontent.com/11884736/198179590-9b4056a4-01f2-4681-94fc-14c57fbfccde.png)
Before volume reduction:
![image](https://user-images.githubusercontent.com/11884736/198180492-a7ba8774-00c9-4b8f-9ea7-5b8ddd67e4be.png)
Stop node 2:
![image](https://user-images.githubusercontent.com/11884736/198180725-567a63d0-0aa7-484e-b6f0-6bbc3571e964.png)
Removing node 2:
![image](https://user-images.githubusercontent.com/11884736/198181171-dfe4fdf4-0256-4e0e-a2f8-d71270a824fd.png)
![image](https://user-images.githubusercontent.com/11884736/198181229-8f64b331-988b-49d0-a990-f21f50d86903.png)
The data migration is complete and no temporary snapshot file is available：
![image](https://user-images.githubusercontent.com/11884736/198181516-880df108-cddf-4773-95a4-872097b31b0c.png)
Query migrated replica data on datanoe1：
![image](https://user-images.githubusercontent.com/11884736/198181720-402ecc99-1ed4-4548-ae2a-85fca8346971.png)










<hr>

This PR has:
    - [ ] concurrent read
    - [x] concurrent write
    - [ ] concurrent read and write 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [x] been tested in a test IoTDB cluster.



<hr>

##### Key changed/added classes (or packages if there are too many classes) in this PR
